diff --git a/src/library.rs b/src/library.rs
index 3f97746..aee8397 100644
--- a/src/library.rs
+++ b/src/library.rs
@@ -1,3 +1,4 @@
+use std::cmp::Reverse;
 use std::collections::HashMap;
 use std::iter::Iterator;
 use std::ops::Deref;
@@ -411,12 +412,13 @@ impl Library {
         }
 
         albums.sort_unstable_by_key(|album| {
-            format!(
-                "{}{}{}",
+            Reverse(format!(
+                "{:?}{}{}{}",
+                album.added_at,
                 album.artists[0].to_lowercase(),
                 album.year,
                 album.title.to_lowercase()
-            )
+            ))
         });
 
         *(self.albums.write().unwrap()) = albums;
